docs/source/asr/datasets.rst

```diff
@@ -190,3 +192,81 @@ in the same directory as the manifest, or even in any specific directory structu
 
 Once you have a manifest that describes each audio file in your dataset, you can then use the dataset by passing
 in the manifest file path in your experiment config file, e.g. as `training_ds.manifest_filepath=<path/to/manifest,json>`.
+
+Tarred Datasets
+---------------
+
+If you are running experiments on a cluster with datasets stored on a distributed file system, you will likely
+want to avoid constantly reading many small files and would prefer tarring your audio files.
+There are tarred versions of some NeMo ASR Dataset classes for this case, such as the ``TarredAudioToCharDataset``
+(corresponding to the ``AudioToCharDataset``) and the ``TarredAudioToBPEDataset`` (corresponding to the
+``AudioToBPEDataset``).
+The tarred audio dataset classes in NeMo use `WebDataset <https://github.com/tmbdev/webdataset>`_.
+
+To use an existing tarred dataset instead of a non-tarred dataset, you will need to set ``is_tarred: true`` in
+your experiment config file.
+Then, you will need to pass in the paths to all of your audio tarballs in ``tarred_audio_filepaths``, either as a list
+of filepaths (e.g. ``['/data/shard1.tar', '/data/shard2.tar']``), or in a single brace-expandable string (e.g.
+``'/data/shard_{1..64}.tar'``).
+
+.. note::
+  For brace expansion, there may be cases where ``{x..y}`` syntax cannot be used due to shell interference.
+  This occurs most commonly inside SLURM scripts. Therefore we provide a few equivalent replacements.
+  Supported opening braces (equivalent to ``{``) are ``(``, ``[``, ``<`` and the special tag ``_OP_``.
+  Supported closing braces (equivalent to ``}``) are ``)``, ``]``, ``>`` and the special tag ``_CL_``.
+  For SLURM based tasks, we suggest the use of the special tags for ease of use.
+
+As with non-tarred datasets, the manifest file should be passed in ``manifest_filepath``.
+The dataloader will assume that the length of the manifest after filtering is the correct size of the dataset
+for reporting training progress.
+
+The ``tarred_shard_strategy`` field of the config file can be set if you have multiple shards and are running
+an experiment with multiprocessing.
+It defaults to ``scatter``, which preallocates a set of shards per node which do not change during runtime.
+
+For more information about the individual tarred datasets and the parameters you can set, including shuffling options,
+see the corresponding class APIs on the `Datasets <./api.html#Datasets>`__ page.
+
+.. warning::
+  If using multiple processes the number of shards should be divisible by the number of workers to ensure an even
+  split among workers.
+  If it is not divisible, logging will give a warning but training will proceed, but likely hang at the last epoch.
+  In addition, if using mutiprocessing, each shard must have the same number of entries after filtering is applied.
```

Comment:
`multinode` vs `multiprocessing`? Or does it even occur under 1 node and multiprocessing. Thats new..

Reply:
In the past I did get it to hang on my local machine with two GPUs (as long as world_size > 1). Though I may be mixing up my PyTorch/Python terminology here... Distributed processing?